kobo/hub/models.py

```diff
@@ -204,23 +204,29 @@ def __str__(self):
         return u"%s" % self.name
 
     def save(self, *args, **kwargs):
+        # save to assign the ID
```

Comment:
I'm not feeling confident about the changes here to call super save() multiple times.

The first thing I note is that https://docs.djangoproject.com/en/4.2/ref/models/instances/#what-happens-when-you-save says that some signals are being emitted during save, so this change seems like it could impact anyone using those signals.

The second concern I have is whether it's safe to assume a DB transaction is being used right now.  Although the kobo settings template in the repo has `ATOMIC_REQUESTS = True`, I'm not sure that means we can assume all kobo users have that enabled.  Some other code in kobo.hub uses `transaction.atomic` which may indicate it was intended to work even if the app sets  `ATOMIC_REQUESTS = False`.

If a transaction is not in progress, it's possible e.g. for the first save() to succeed and then the request to be interrupted before the rest.  In the rare worst case, with the recently added create-worker command, it might successfully create a Worker object but not assign a worker_key (i.e. passes the first save() here and then dies).  Retrying the create-worker command in that case wouldn't be able to recover from this situation as it would try to create a new Worker with the same name.

I understand the point here is to avoid `Task.objects.opened().filter(worker=self)` if the worker's ID isn't assigned yet.  I believe if the worker has no ID (we are creating it) then it couldn't possibly have any tasks yet, so what do you think about keeping just one call to super save() and not querying for tasks at all in that case, something like:

```
tasks = Task.objects.opened().filter(worker=self) if self.id else []
self.task_count = tasks.count() if tasks else 0
self.current_load = [...] # I think all the rest still works just fine
```

Reply:
Thank you for the explanation, @rohanpm!  I've pushed a new revision that only saves once.  Originally, I thought it is fine to save twice because `kobo` already does that in the `Task.save` method (introduced in c5d7f8e68a24df08a5425206ae686c34f0ebf135):
https://github.com/release-engineering/kobo/blob/ac6042419363747b2618c80cfb5d5909ae4ad8d0/kobo/hub/models.py#L601-L603
I guess that should be fixed as well, right?

Reply:
Yes, `Task.save` does also look dubious, if we don't know that a transaction is in progress then we can't be sure all the individual saves there will always happen (or not) together.

Reply:
I'm made a separate PR for `Task.save`: https://github.com/release-engineering/kobo/pull/221